plugins/early_talker.js

```diff
@@ -41,6 +47,12 @@ exports.early_talker = function(next, connection) {
         return next();
     }
 
+    // Don't delay whitelisted IPs
+    if (plugin.whitelist && plugin.ip_in_list(plugin.whitelist, connection.remote_ip)) { // check connecting IP
```

Comment:
There's no need to pass plugin.whitelist here as an argument. Just pass in the remote_ip. In exports.ip_in_list, you can add a `var plugin = this` at the beginning and you'll have access to plugin.whitelist within the function.